File: python/LeetCode_1913_MaximumProductDifferenceBetweenTwoPairs.py
def maxProductDifference(nums) -> int: 
    max_m = 0
    max_m_m = max_m 
    min_m = 10000
    min_m_m = min_m
    
    for n in nums:
        if n > max_m:
            max_m_m = max_m
            max_m = n         
        elif n > max_m_m or max_m == n:
            max_m_m = n

        if n < min_m:
            min_m_m = min_m 
            min_m = n
        elif n < min_m_m or n == min_m:
            min_m_m = n

    return (max_m * max_m_m) - (min_m * min_m_m)


print(maxProductDifference([4,2,5,9,7,4,8]))

File: c/LeetCode_1913_MaximumProductDifferenceBetweenTwoPairs.c
#include <stdio.h>
#include <stdlib.h>


int maxProductDifference(int* nums, int numsSize){
    int max = 0;
    int max_m = max; 
    int min = 10000;
    int min_m = min;

    for (int i = 0; i < numsSize; ++i) {
        if (max < nums[i]) {
            max_m = max;
            max = nums[i];
        } else if (nums[i] > max_m || nums[i] == max) {
            max_m = nums[i];
        }

        if (min > nums[i]) {
            min_m = min;
            min = nums[i];
        } else if (nums[i] < min_m || nums[i] == min) {
            min_m = nums[i];
        }
    }

    return (max * max_m) - (min * min_m);
}


int main() {
    return 0;
}